farmerbot/parser/validator.go

```diff
@@ -0,0 +1,88 @@
+package parser
+
+import (
+	"fmt"
+	"slices"
+
+	substrate "github.com/threefoldtech/tfchain/clients/tfchain-client-go"
+	"github.com/threefoldtech/tfgrid-sdk-go/farmerbot/internal"
+)
+
+// wrapper for validateInput
+func ValidateConfig(input internal.Config, network string) error {
+	manager := substrate.NewManager(internal.SubstrateURLs[network]...)
+	subConn, err := manager.Substrate()
+	if err != nil {
+		return err
+	}
+	defer subConn.Close()
+	return validateInput(input, subConn)
+}
+
+// validateInput validates that included, excluded and priority nodes are in the farm
+func validateInput(input internal.Config, sub internal.Substrate) error {
+	nodes, err := sub.GetNodes(input.FarmID)
+	if err != nil {
+		return fmt.Errorf("couldn't retrieve node for %d : %v", input.FarmID, err)
+	}
+	nodesMap := make(map[uint32]bool)
+	for _, node := range nodes {
+		nodesMap[node] = true
+	}
+	includedNodes := make(map[uint32]bool)
+	if len(input.IncludedNodes) != 0 {
+		if err := validateIncludedNodes(input.IncludedNodes, input.ExcludedNodes, nodesMap); err != nil {
+			return err
+		}
+		for _, includedNode := range input.IncludedNodes {
+			includedNodes[includedNode] = true
+		}
+	} else {
+		for key, value := range nodesMap {
+			if slices.Contains(input.ExcludedNodes, key) {
+				continue
+			}
+			includedNodes[key] = value
+		}
+	}
```

Comment:
```suggestion
        for _, includedNode := range input.IncludedNodes {
		includedNodes[includedNode] = true
	}
	if len(icludedNodes == 0){
	       for key, value := range nodesMap {
			if !slices.Contains(input.ExcludedNodes, key) {
				includedNodes[key] = value
			}
		}
	}

	if err := validateIncludedNodes(input.IncludedNodes, input.ExcludedNodes, nodesMap); err != nil {
		return err
	}

```

Reply:
validateIncludedNodes will be called in all cases kda, it need to be in else statment right?